File: algorithm/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int curr= 0, n = nums.size();
        for (int i=0; i<n; ++i){
            if (nums[i]!= 0){ 
                nums[curr]= nums[i];
                curr++;
            }
        }
        while (curr<n)
            nums[curr++]= 0;
    }
};

File: python/array and string/move-zeroes.py
#https://leetcode.com/problems/move-zeroes/

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        curr = 0
        for i in range(len(nums)):
            if nums[i]:
                nums[i], nums[curr] = nums[curr], nums[i]
                curr+= 1
        
        return ""